### PR TITLE
Remove XE/XK documentation, tests, and module build infrastructure

### PR DESCRIPTION
**Pull request by ronawho**
XE/XK are no longer officially supported, so remove docs/tests for them. I left
in the ugni code and the chpl_platform.py code, so technically you can still
build and run on XE, but we're no longer testing this configuration. I added
compiler and makefile warning to mention this, and plan on removing the
remaining ugni/chplenv support after the next release. We could probably just
remove it now, but there's really not much XE specific code, so I don't think
it hurts to keep around for little longer.

Note that I did have to remove some named args for calls to .decode() in our
build scripts to make the python 2.6 compatible again. This was added in #15652

Closes https://github.com/cray/chapel-private/issues/327

### CODE REVIEW

**Comment by ronawho** [comment]
@e-kayrakli mostly pinging you for the second commit, feel free to ignore the test.

**Comment by ronawho** [comment]
@gbtitus here's a prototype of removing gemini support from ugni-- https://github.com/ronawho/chapel/commit/22b125accda3eec83a04b54ab6388075d3594926

I don't mind keeping that in there for the next release and then ripping it out, but I would also have no problem removing that now. Let me know if you have a preference.

File: doc/rst/platforms/cray.rst [hunk 1]
```diff
@@ -134,8 +130,8 @@ built from source on Cray CS systems using the
 Building Chapel for a Cray System from Source
 ---------------------------------------------
 
-1) If using a XC, XE, or XK system, continue to step 2. If using a
-   CS series system, set ``CHPL_HOST_PLATFORM`` to ``cray-cs``.
+1) If using a XC system, continue to step 2. If using a CS series
```
Comment (gbtitus) [inline doc/rst/platforms/cray.rst:133]: Think this should be (should always have been) "an" XC system, because of the initial vowel phoneme.

File: doc/rst/platforms/cray.rst [hunk 2]
```diff
@@ -473,10 +457,10 @@ To use ugni communications:
 Network Atomics
 _______________
 
-The Gemini and Aries networks on Cray XE, XK, and XC series systems
-support remote atomic memory operations (AMOs).  When the
-``CHPL_NETWORK_ATOMICS`` environment variable is set to ``ugni``, the
-following operations on remote atomics are done using the network::
+The Aries networks on Cray  XC series systems support remote atomic
```
Comment (gbtitus) [inline doc/rst/platforms/cray.rst:460]: Looks like there's an extra space between "Cray" and "XC".

File: doc/rst/technotes/localeModels.rst [hunk 3]
```diff
@@ -104,14 +104,14 @@ when using the flat locale model, and often worse.  The *multi-ddata*
...
 the end we disabled it because much of the performance loss was inherent
-in the implementation and could not be removed.  On Cray XE and XC
-systems with ``CHPL_COMM=ugni`` and NIC-registered memory, recent work
-to allocate arrays separately and register them dynamically has improved
-NUMA affinity and thus performance, but the benefits of that effort
-apply to the flat locale model as well as they do to the numa one.  For
-other configurations, with the multi-ddata feature disabled performance
-with the NUMA locale model has returned, for better or worse, to what it
-was before that was introduced.  At present most of our effort has to do
+in the implementation and could not be removed.  On Cray XC systems with
```
Comment (gbtitus) [inline doc/rst/technotes/localeModels.rst:107]: The multi-ddata feature is not just disabled as we say below, but removed and won't return.  I think we could simply remove the two sentences "The multi-ddata feature ... and could not be removed." here, and also the last two in this paragraph "For other configurations, ... first-touch to achieve NUMA affinity."

File: test/runtime/configMatters/comm/ugni/overflow-nic-tlb.wrap [hunk 4]
```diff
@@ -11,8 +11,6 @@ if [[ -n "$hpm" ]] ; then module unload $hpm ; fi
 
 export CHPL_RT_CALL_STACK_SIZE=128K
 case $($CHPL_HOME/util/chplenv/chpl_platform.py --target) in
```
Comment (gbtitus) [inline test/runtime/configMatters/comm/ugni/overflow-nic-tlb.wrap:13]: Maybe just remove the `case` statement entirely? Without a `*)` it doesn't seem needed.

File: util/build_configs/cray-internal/README.md [hunk 5]
```diff
@@ -6,7 +6,7 @@ one of the setenv scripts in this subdir, setenv-example-3 from the parent direc
 or a new setenv script derived from one of those. See `chapel_build.bash` for illustration.
 
 The Chapel RPM file produced by these scripts is intended to be fully compatible
-with current generation Cray-XC/XE supercomputers and system-management tools.
+with current generation Cray-XC supercomputers and system-management tools.
```
Comment (gbtitus) [inline util/build_configs/cray-internal/README.md:9]: no hyphen in "Cray XC"

File: util/build_configs/cray-internal/README.md [hunk 6]
```diff
@@ -57,7 +57,6 @@ new subdir.
   are Cray-specific implementations of the general build_configs/setenv pattern:
   - setenv-xc-x86_64:  Chapel Cray-XC module for x86_64
```
Comment (gbtitus) [inline util/build_configs/cray-internal/README.md:58]: No hyphen in "Cray XC".  This is actually a general flaw in `$CHPL_HOME/util` which I should have caught in the beginning.  It could be fixed separately.

File: util/chplenv/utils.py [hunk 7]
```diff
@@ -46,13 +46,13 @@ def run_command(command, stdout=True, stderr=False, cmd_input=None):
     except OSError:
         error("command not found: {0}".format(command[0]), OSError)
 
-    byte_cmd_input = str.encode(cmd_input, encoding="utf-8") if cmd_input else None
+    byte_cmd_input = str.encode(cmd_input, "utf-8") if cmd_input else None
```
Comment (gbtitus) [inline util/chplenv/utils.py:49]: Are these changes maybe part of some other PR?  (Don't seem XE/XK related.)

File: util/config/make_sys_basic_types.py [hunk 8]
```diff
@@ -102,7 +102,7 @@ def get_sys_c_types(docs=False):
     compileline_env.pop('CHPL_MAKE_CHPLENV_CACHE', None)
     compileline_proc = subprocess.Popen([compileline_cmd, '--compile'],
         stdout=subprocess.PIPE, env=compileline_env)
-    compileline = compileline_proc.communicate()[0].decode(encoding="utf-8").strip();
+    compileline = compileline_proc.communicate()[0].decode("utf-8").strip();
```
Comment (gbtitus) [inline util/config/make_sys_basic_types.py:105]: Not part of this PR?

File: util/chplenv/utils.py (hunk 7 above)
Comment (gbtitus) [reply util/chplenv/utils.py:49]: Oops! my bad -- this is mentioned in the PR text.  Self-resolving ...

File: util/build_configs/cray-internal/README.md (hunk 5 above)
Comment (ronawho) [reply util/build_configs/cray-internal/README.md:9]: These scripts are internal enough that I'm not sure it's worth correcting.

File: util/build_configs/cray-internal/README.md (hunk 6 above)
Comment (ronawho) [reply util/build_configs/cray-internal/README.md:58]: In this case I think `Cray-XC` could be right because it's talking about our module, and not the platform as a whole. Either way I'm not really sure these are worth addressing.

**Comment by ronawho** [comment]
I think I've addressed all feedback, except the Cray-XC hypen comments.

**Comment by gbtitus** [comment]
> I think I've addressed all feedback, except the Cray-XC hypen comments.

I'm happy to put on my "Mr. Nitpicky" vest and do those.

**Comment by ronawho** [comment]
After discussion, we've decided to remove the ugni support as well. Closing this in favor of https://github.com/chapel-lang/chapel/pull/15841 which does that.
